client/asset/btc/btc.go

```diff
@@ -1464,23 +1524,55 @@ func (btc *baseWallet) connect(ctx context.Context) (*sync.WaitGroup, error) {
 // Connect connects the wallet to the btc.Wallet backend and starts monitoring
 // blocks and peers. Satisfies the dex.Connector interface.
 func (btc *intermediaryWallet) Connect(ctx context.Context) (*sync.WaitGroup, error) {
-	wg, err := btc.connect(ctx)
+	// A separate baseCtx is used to ensure that all subprocesses are complete
+	// before the node is shut down.
+	baseCtx, killBase := context.WithCancel(context.Background())
```

Comment:
This pattern is a little scary. Lemme see if I follow the reasoning. Basically, we want any process that might write to the DB to complete before we `Close` the DB. If we didn't do this, we might potentially see errors in the logs about trying to write to a closed db. Is that right? How likely are those errors? Do they matter?

Reply:
This is not just for the DB, but also for the client. There were errors in which the client was already shut down, but it was being called. I guess the errors may not matter, but imo it's more scary to be running code that calls things which are already shut down.

Reply:
The more I look at this the more I don't like it. If the context is canceled, we shouldn't be trying to use the wallet anyway. Are the various goroutines checking the context at the appropriate places? A stray context canceled error is ok, but if it happens often, we can probably resolve it without this background-derived context.

Reply:
OK I've removed the double wg from all wallets. There should only be some context cancelled/db already closed errors.